### PR TITLE
Enable MathJax and futz with variable names

### PR DESCRIPTION
**Pull request by jedbrown**

### CODE REVIEW

File: src/rev.md [hunk 1]
```diff
@@ -19,58 +19,63 @@ Both the inplace and "normal" variant return the gradient. The difference is tha
 ### Usage story
 Let us start by looking at the most basic examples we can think of:
 
-$f(x,y) = x^2 + y*3$
+\\[ f(x,y) = x^2 + 3y \\]
 
-We have two input variables $x$, $y$ and a scalar return value.  Just to check our sanity, we first pass it to [wolfram alpha](https://www.wolframalpha.com/input?i2d=true&i=D%5BPower%5Bx%2C2%5D+%2B+y*3%2Cx%5D%3B+D%5BPower%5Bx%2C2%5D+%2B+y*3%2Cy%5D%3B+). No big surprises so far. Let's check for Enzyme (our compiler explorer does not handle Rust yet, so you'l have to trust me on this).
+We have two input variables \\(x\\), \\(y\\) and a scalar return value.  Just to check our sanity, we first pass it to [wolfram alpha](https://www.wolframalpha.com/input?i2d=true&i=D%5BPower%5Bx%2C2%5D+%2B+y*3%2Cx%5D%3B+D%5BPower%5Bx%2C2%5D+%2B+y*3%2Cy%5D%3B+). No big surprises so far. Let's check for Enzyme (our compiler explorer does not handle Rust yet, so you'l have to trust me on this).
```
Comment (jedbrown) [inline src/rev.md:24]: mdBook doesn't support `$x$` notation. https://rust-lang.github.io/mdBook/format/mathjax.html

File: src/rev.md [hunk 2]
```diff
@@ -19,58 +19,63 @@ Both the inplace and "normal" variant return the gradient. The difference is tha
...
 fn f(x: f32, y: f32) -> f32 {
-  x*x + 3.0 * y
+  x * x + 3.0 * y
 }
 fn df(x: f32, y: f32) -> (f32, f32, f32) {
-  (2.0 * x, 3.0, x*x + 3.0 * y)
+  let d_dx = 2.0 * x;
+  let d_dy = 3.0;
+  let f = x * x + 3.0 * y;
+  (d_dx, d_dy, f)
```
Comment (jedbrown) [inline src/rev.md:43]: I think it's more clear to name these variables rather than explain a big tuple in text.

I personally think of `dx` as being a variation in `x`, which is correct for forward mode. In reverse mode, the notation is usually $\bar x$, which is shorthand for $\partial (\cdot)/\partial x$ where the functional $(\cdot)$ is implicit and assumed to be clear from context. I'm happy to use other naming here, but I think using `dx` for this quite different object is confusing.

File: src/rev.md [hunk 3]
```diff
@@ -19,58 +19,63 @@ Both the inplace and "normal" variant return the gradient. The difference is tha
...
+We could obviously zero-initialize a float for the user, but let's assume the constructor is complex due to involving a double-linked-list or ffi, so we can't guarantee that on the compiler side. As an alternative motivation, imagine that we call `df` 5 times in a loop. It is clear that in this case the accumulated gradients should be 5 times higher too, which won't happen if we set `d_dy = 3.0` each time, instead of using `+=`. Yet another reason would be higher-order derivatives (todo: just refer to literature?).
 
 Now that we got back from this rabbit hole, let's go wild and train a neural network on our local national lab server:
 
 ```rust
 #[autodiff(backprop, Reverse, Duplicated, Duplicated, Active)]
-fn inference(images: &[f32], weights: &[f32]) -> f32 {
+fn training_loss(images: &[f32], weights: &[f32]) -> f32 {
```
Comment (jedbrown) [inline src/rev.md:78]: I just can't stomach "inference" for a model that predicts observables. (Before the self-aggrandizing ML terminology entered the lexicon with the opposite meaning, there was consensus that the word refers to estimating latent quantities.) But anyway, this is a loss function, not just evaluating the forward `predict` (that ML people call inference). Technically, we would have labels or known quantities also as input to the loss function, but we can gloss over that.

File: src/rev.md (hunk 1 above)
Comment (ZuseZ4) [reply src/rev.md:24]: nice, I didn't scroll down that far and was already looking for plugins yesterday

File: src/rev.md (hunk 3 above)
Comment (ZuseZ4) [reply src/rev.md:78]: fair enough

**Review by ZuseZ4 — APPROVED** [review]
